libraries/operation-graph/src/WatchLoop.ts

```diff
@@ -192,9 +192,12 @@ export class WatchLoop implements IWatchLoopState {
 
             try {
               status = await this.runUntilStableAsync(abortController.signal);
-              // ESLINT: "Promises must be awaited, end with a call to .catch, end with a call to .then ..."
-              // eslint-disable-next-line @typescript-eslint/no-floating-promises
-              this._requestRunPromise.finally(requestRunFromHost);
+              this._requestRunPromise
+                // the reject callback in the promise is discarded so we ignore errors
+                .catch(() => {})
```

Comment:
Is this behavior correct/expected?

Reply:
the watchloop is using a deferred promise pattern, but only binds the `resolve` function per:
https://github.com/microsoft/rushstack/blob/ca6e96bcaac381ffe078dd5188b81599f0dadec0/libraries/operation-graph/src/WatchLoop.ts#L264-L266

Reply:
@iclanton ?

Reply:
The logic here is wrong. `requestRunPromise` resolves with the `requestor`, which is why it had directly passed `requestRunFromHost` as the callback.